src/components/Workspace.jsx

```diff
@@ -188,6 +298,7 @@ export default class Workspace extends React.Component {
             {this._renderEnvironment()}
           </div>
         </div>
+        {this._renderFullyMinimizedColumns()}
```

Comment:
A couple of thoughts on this. First, in the case where one column is totally collapsed, I think we would ideally put all the hidden components together in the DOM—so basically I would move this up to right below the call to `_renderHiddenRightColumnComponents()`.

Second, I would propose rearranging the logic a little bit:

* Remove the check for “all left column components hidden” from `_renderHiddenLeftColumnComponents()`, and inline `_renderHiddenLanguages()`. The method already won’t be called if all the components are hidden, because of the check on line 238.
* Get rid of the `_renderFullyMinimizedColumns()` method (I didn’t find “fully minimized” an intuitive description on the first read-through of the code) and just inline a check here (or wherever this line ends up), i.e. `{!this._shouldRenderLeftColumn() && this._renderHiddenLeftColumnComponents()}`